src/libs/middleware.ts

```diff
@@ -0,0 +1,29 @@
+import { IncomingMessage, ServerResponse } from "http";
+import { User, GatekeeperRequestError } from "../common/UserProvider";
+import { getBaseUrl, authenticatedFetch } from "./fetcher";
+
+type Request = IncomingMessage & { cookies: { accessToken: string } };
+
+type AuthenticatedRouteHandler = (
+  req: IncomingMessage & any,
+  res: ServerResponse & any,
+  user: User
+) => any;
+
+export const authenticatedRoute = (
+  handler: AuthenticatedRouteHandler
+) => async (req: Request, res: ServerResponse): Promise<any> => {
```

Comment:
```suggestion
) => async (req: Request, res: ServerResponse): Promise<void> => {
```

I think you could get away with setting the return value to Promise<void>

Reply:
Had it as `any` to handle any handler response. Changed to void